### PR TITLE
Withdraw function with IERC20 Dispatcher transfer

### PR DESCRIPTION
**Pull request by Pvsaint**
feat: implemented and tested withdrawal function, implemented ERC20 token transfer and interface

### CODE REVIEW

**Comment by Pvsaint** [comment]
I closed the previous PR and opened a new one

**Comment by mubarak23** [comment]
> I closed the previous PR and opened a new one

both build and test need to pass, at the moment they are all failing

**Comment by Pvsaint** [comment]
Fixed

**Comment by Pvsaint** [comment]
Everything should work fine now sir!

File: src/campaign.cairo [hunk 1]
```diff
@@ -142,20 +195,38 @@ mod TokengiverCampaign {
...
+        fn withdraw(ref self: ContractState, campaign_address: ContractAddress, amount: u256) {
+            let campaign: Campaign = self.campaign.read(campaign_address);
+            let caller: ContractAddress = get_caller_address();
+
+            assert(caller == campaign.campaign_owner, NOT_CAMPAIGN_OWNER);
+
+            let available_balance: u256 = self.withdrawal_balance.read(campaign_address);
+            assert(amount <= available_balance, INSUFFICIENT_BALANCE);
+
+            self.withdrawal_balance.write(campaign_address, available_balance - amount);
```
Comment (mubarak23) [inline src/campaign.cairo:207]: this should go after transfer, only run this when the transaction , 

check to ensure the transfer was successful via ASSERTION, once it is true,

`self.withdrawal_balance.write(campaign_address, available_balance - amount);`

**Review by mubarak23 — CHANGES_REQUESTED** [review]
Also write unit test for the function

**Pull request closed without merging**